File: services/intelligence-api/app/schemas/decision.py
"""Decision schemas — recommendations, pressure scoring, and explanations."""
from pydantic import BaseModel, Field
from enum import Enum
from .scenario import GDPComponentCode, SectorCode, LanguageVariant


class DecisionUrgency(str, Enum):
    IMMEDIATE = "immediate"
    SHORT_TERM = "short_term"
    MEDIUM_TERM = "medium_term"
    LONG_TERM = "long_term"


class DecisionPressure(BaseModel):
    """Quantified decision pressure with stakeholder context."""
    score: float = Field(ge=0.0, le=100.0)
    urgency: DecisionUrgency
    affected_stakeholders: list[str]
    primary_drivers: list[str]


class DecisionRecommendation(BaseModel):
    """A single actionable decision recommendation."""
    id: str
    scenario_id: str
    title: LanguageVariant
    action: LanguageVariant
    priority: DecisionUrgency
    confidence: float = Field(ge=0.0, le=1.0)
    affected_entities: list[str]
    pressure: DecisionPressure


class DecisionExplanation(BaseModel):
    """Structured explanation for why a decision was recommended."""
    decision_id: str
    what_happened: LanguageVariant
    why_it_matters: LanguageVariant
    who_is_affected: list[str]
    gdp_components_moved: list[GDPComponentCode]
    sectors_under_pressure: list[SectorCode]
    why_this_recommendation: LanguageVariant
    likely_next_developments: list[LanguageVariant]
    confidence: float = Field(ge=0.0, le=1.0)


class ExplainedDecision(BaseModel):
    """Decision + full explanation bundle."""
    recommendation: DecisionRecommendation
    explanation: DecisionExplanation

File: apps/executive-dashboard/src/types/decision.ts
/**
 * Decision Types — exact mirror of schemas/decision.py
 *
 * Source: services/intelligence-api/app/schemas/decision.py
 */

import type { LanguageVariant, GDPComponentCode, SectorCode } from "./scenario";

export type DecisionUrgency = "immediate" | "short_term" | "medium_term" | "long_term";

export interface DecisionPressure {
  score: number; // 0-100
  urgency: DecisionUrgency;
  affected_stakeholders: string[];
  primary_drivers: string[];
}

export interface DecisionRecommendation {
  id: string;
  scenario_id: string;
  title: LanguageVariant;
  action: LanguageVariant;
  priority: DecisionUrgency;
  confidence: number; // 0-1
  affected_entities: string[];
  pressure: DecisionPressure;
}

export interface DecisionExplanation {
  decision_id: string;
  what_happened: LanguageVariant;
  why_it_matters: LanguageVariant;
  who_is_affected: string[];
  gdp_components_moved: GDPComponentCode[];
  sectors_under_pressure: SectorCode[];
  why_this_recommendation: LanguageVariant;
  likely_next_developments: LanguageVariant[];
  confidence: number; // 0-1
}

export interface ExplainedDecision {
  recommendation: DecisionRecommendation;
  explanation: DecisionExplanation;
}
